### PR TITLE
On Windows, verify browsers installed before launching.

### PR DESCRIPTION
**Pull request by TimBarham**
Do this to avoid dialog that pops up on Windows if browser is not installed, and provide more meaningful error messages.

### CODE REVIEW

File: cordova-serve/src/browser.js [hunk 1]
```diff
@@ -17,9 +17,15 @@
...
  */
 
-var exec = require('./exec'),
+var child_process = require('child_process'),
+    exec = require('./exec'),
+    fs = require('fs'),
+    path = require('path'),
     Q = require('q');
 
+var NOT_INSALLED = 'The browser target is not installed: %target%';
```
Comment (vladimir-kotikov) [inline cordova-serve/src/browser.js]: typo: `NOT_INSTALLED`

File: cordova-serve/src/browser.js [hunk 2]
```diff
@@ -72,7 +79,10 @@ module.exports = function (opts) {
             args.push(url);
         }
         var command = args.join(' ');
-        return exec(command);
+        return exec(command).catch(function (error) {
+            // Assume any error means that the browser is not installed and display that as a more friendly error.
+            throw new Error(NOT_INSALLED.replace('%target%', target));
```
Comment (vladimir-kotikov) [inline cordova-serve/src/browser.js]: typo again

File: cordova-serve/src/browser.js (hunk 1 above)
Comment (TimBarham) [inline cordova-serve/src/browser.js]: Thanks. Oops. Fixed :smile:

File: cordova-serve/src/browser.js [hunk 3]
```diff
@@ -95,16 +105,93 @@ function getBrowser(target, dataDir) {
...
+    var promise = target === 'edge' ? edgeSupported() : browserInstalled(browser);
+    return promise.catch(function (error) {
+        return Q.reject((error && error.toString() || NOT_INSTALLED).replace('%target%', target));
+    });
+}
+
+function edgeSupported() {
+    var d = Q.defer();
+
+    child_process.exec('ver', function (err, stdout, stderr) {
```
Comment (vladimir-kotikov) [inline cordova-serve/src/browser.js]: Is there any chance to reuse `./exec` module here. Looks like it contains almost the same logic.

Comment (vladimir-kotikov) [inline cordova-serve/src/browser.js]: BTW, `os.version()` gives me exactly the same version as `ver`, it might better to use it

Comment (TimBarham) [inline cordova-serve/src/browser.js]: I originally used the exec module and the logic actuallly ended up simpler this way.

Comment (TimBarham) [inline cordova-serve/src/browser.js]: os.release() doesn't recognize Win10 in fairly common versions of node.

**Comment by vladimir-kotikov** [comment]
Looks good.

**Comment by TimBarham** [comment]
Thanks for taking a look, @vladimir-kotikov.

**Pull request merged**